src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppExtensions.cs

```diff
@@ -31,4 +43,266 @@ public static IDistributedApplicationBuilder AddAzureContainerAppsInfrastructure
+                        EnabledProtocol = FileShareEnabledProtocol.Smb
+                    };
+
+                    infra.Add(share);
+
+                    var keysExpr = storageVolume.GetKeys()[0].Compile();
+                    var keyValue = new MemberExpression(keysExpr, "value");
+
+                    var containerAppStorage = new ContainerAppManagedEnvironmentStorage(managedStorageName)
+                    {
+                        Parent = containerAppEnvironment,
+                        ManagedEnvironmentStorageAzureFile = new()
+                        {
+                            ShareName = share.Name,
+                            AccountName = storageVolume.Name,
+                            AccountKey = keyValue,
+                            AccessMode = ContainerAppAccessMode.ReadWrite
+                        }
+                    };
+
+                    infra.Add(containerAppStorage);
+
+                    managedStorages[outputName] = containerAppStorage;
+                }
+            }
+
+            // REVIEW: These don't need to be coupled to the container app environment
+            var kvs = new Dictionary<string, KeyVaultService>();
+
+            foreach (var (key, _) in resource.SecretKeyVaultNames)
+            {
+                var kvIdentifier = Infrastructure.NormalizeBicepIdentifier($"kv-{key}");
+                var kv = new KeyVaultService(kvIdentifier)
+                {
+                    Properties = new()
+                    {
+                        TenantId = BicepFunction.GetTenant().TenantId,
+                        Sku = new KeyVaultSku()
+                        {
+                            Family = KeyVaultSkuFamily.A,
+                            Name = KeyVaultSkuName.Standard
+                        },
+                        EnableRbacAuthorization = true
+                    },
+                    Tags = tags,
+                };
+
+                infra.Add(kv);
+
+                // Role assignments
+                var kvra = kv.CreateRoleAssignment(KeyVaultBuiltInRole.KeyVaultAdministrator, identity);
+                kvra.Name = BicepFunction.CreateGuid(kv.Id, identity.Id, kvra.RoleDefinitionId);
+                infra.Add(kvra);
+
+                kvs[key] = kv;
+            }
+
+            // Add the volume outputs to the container app environment storage
+            foreach (var (key, value) in managedStorages)
+            {
+                infra.Add(new ProvisioningOutput(key, typeof(string))
+                {
+                    Value = value.Name
+                });
+            }
+
+            // Add the secret outputs to the key vaults
+            foreach (var (key, value) in kvs)
+            {
+                infra.Add(new ProvisioningOutput(key, typeof(string))
+                {
+                    Value = value.Name
+                });
+            }
+
+            infra.Add(new ProvisioningOutput("MANAGED_IDENTITY_NAME", typeof(string))
```

Comment:
Looks like you missed `MANAGED_IDENTITY_CLIENT_ID`

Reply:
Do we still use this after your change?

Reply:
We don't this comes from the per container app identity now.